src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -350,63 +357,83 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
+     *                     severed text node.
+     */
+    fluid.prefs.enactor.syllabification.insertIntoTextNode = function (node, toInsert, position) {
+        node = node.splitText(position);
+        node.parentNode.insertBefore(toInsert, node);
+        return node;
+    };
+
+    /**
+     * Insert separators, indicating the hyphenated positions, into a text node. This will inject the specified
+     * markup as the separators and split the original textnode at the hyphenation points. If there are no places to
+     * inject a hyphenation point, the node will be left unchanged. If the original textnode inclues soft hyphens (i.e.
+     * &shy;) characters, these will be used as a hyphenation point with the original location in the text node
+     * replacded with the `softHyphenPlaceholderMarkup` to allow it to be restored when syllabification is removed.
      *
      * @param {HypherHyphenator} hyphenator - an instance of a Hypher Hyphenator
      * @param {DomNode} node - a DOM node containing text to be hyphenated
      * @param {String} separatorMarkup - the markup to be injected and used to indicate the separators/hyphens
+     * @param {String} softHyphenPlaceholderMarkup - the markup to be injected and used to indicate the original
+     *                                               location of any soft hyphens included in `node`
      */
-    fluid.prefs.enactor.syllabification.hyphenateNode = function (hyphenator, node, separatorMarkup) {
-        if (!hyphenator) {
+    fluid.prefs.enactor.syllabification.hyphenateNode = function (hyphenator, node, separatorMarkup, softHyphenPlaceholderMarkup) {
+        if (!hyphenator || !node.textContent) {
             return;
         }
 
-        // remove \u200B characters added hyphenateText
-        var hyphenated = hyphenator.hyphenateText(node.textContent).replace(/\u200B/gi, "");
+        var hyphenated = hyphenator.hyphenateText(node.textContent);
 
         // split words on soft hyphens
         var segs = hyphenated.split("\u00AD");
+
         // remove the last segment as we only need to place separators in between the parts of the words
         segs.pop();
+
         fluid.each(segs, function (seg) {
-            // Do not attempt to process empty segs, which could lead to an infinite recursion.
+            // If the text content starts with a soft hyphen (\u00AD, &shy;) replace it with an empty placeholder; which
+            // is used to put the soft hyphen back if syllabification is disabled.
             // See: https://issues.fluidproject.org/browse/FLUID-6554
-            if (seg.length) {
-                var separator = $(separatorMarkup)[0]; // converts the separatorMarkup markup string into a Dom Node
-                node = node.splitText(seg.length);
-                node.parentNode.insertBefore(separator, node);
+            if (node.textContent.startsWith("\u00AD")) {
```

Comment:
One problem is, when a soft hyphen is in a middle of a word, syllabification breaks the word into 2 syllables at where the soft hyphen is regardless if these 2 sections are proper syllables. 

For example, `back` is one single syllable. When a soft hyphen is in between `ba` and `ck`, hypher will mark them as 2 separate syllables, which might be misleading.

I was thinking about removing soft hyphens from texts before running them thru the hypher and inserting soft hyphens back. But the difficult part is how to keep track of positions of those soft hyphens.

Reply:
One suggestion I can think of is to simply clone any DOM that the soft hyphens are removed from and hold the original, together with a pointer to its parent, it in a private fragment out of the document context. This can then be swapped back in again when the enactor is turned off.

Reply:
Removing soft hyphens without inserting them back when the "syllable" setting is on will lead to the issue below that hyphenations are gone.

Reply:
@cindyli I can see your point. However, the ability to break on soft hyphens provides a means for an integrator to define where the syllables are split. Unfortunately the hyphenation engine we use it's 100% perfect. If I remember correctly, it's closer to 80% accurate. So providing an ability for an integrator to explicitly define where breaks should happen may be useful. Also for novel words or localization specific pronunciation that we may not cover. 

I'd suggest not using soft hyphens where a word wouldn't have a natural hyphenation point. Although I'm not sure if there are reasons that would contradict this nor is it something we could enforce in anyway. 

More thoughts/comments welcome.

Reply:
I think it's reasonable to imagine that turning on the syllabification enactor represents the user's intention "I would like to substitute UIO's idea for where syllables should break for the original page author's, if any". So long as we restore the markup to its original condition when the enactor is deactivated, I think this will be accepted. Of course it would be desirable if we could somehow incorporate the "hint" from the markup author provided by the soft hyphen but this doesn't seem to be possible with hypher.